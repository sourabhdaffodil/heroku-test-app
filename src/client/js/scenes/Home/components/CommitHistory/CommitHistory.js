
import React, { Component } from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import CommitsGroup from '../CommitsGroup/CommitsGroup';
import { media } from '../../../../theme/style-utils';
import Icon from '../../../../components/Icon/Icon';

const CommitListing = styled.div`
  position: relative;
  
  &::before {
    position: absolute;
    top: 10px;
    bottom: -10px;
    left: 0;
    z-index: -1;
    display: block;
    width: 1px;
    content: "";
    background-color: ${props => props.theme.color('background')};
    margin-top: 20px;
    
    ${media.phone`
      display: none;
    `}
  }

  &:last-child ul {
    ${media.phone`
      border-bottom: 0;
    `}
  }
`;

const Head = styled.div`
  display: flex;
  align-items: center;
  padding-left: 20px;
  
  ${media.phone`
    padding-left: 0;
  `}

  i.fa-circle-o {
    position: absolute;
    left: -6px;
    color: ${props => props.theme.color('background')};

    ${media.phone`
      display: none;
    `}
  }
`;

const Title = styled.div`
  padding: 7px 0;
  color: ${props => props.theme.contrast('light', 'secondary')};
  flex-grow: 1;
  ${props => props.theme.textSetting('sm')};
  
  ${media.phone`
    padding: 7px 15px;
    font-weight: ${props => props.theme.weight('semiBold')};
    background-color: ${props => props.theme.color('background', 'light')};
    ${props => props.theme.textSetting('sm')};
  `}

  span {
    ${media.phone`
      display: none;
    `}
  }
`;

class CommitHistory extends Component {
  renderHistory() {
    const years = Object.keys(this.props.commits).reverse();

    return years.map(year => (
      <CommitListing key={year}>
        <Head>
          <Icon icon="circle-o" />
          <Title><span>Commits on </span>{year}</Title>
        </Head>
        <CommitsGroup commits={this.props.commits[year]} />
      </CommitListing>
    ));
  }

  render() {
    return (
      <div>
        {this.renderHistory()}
      </div>
    );
  }
}

CommitHistory.propTypes = {
  commits: PropTypes.object.isRequired,
};

function mapStateToProps(state) {
  return {
    commits: state.history.commits,
  };
}

export default connect(mapStateToProps)(CommitHistory);
export {
  CommitHistory,
  mapStateToProps,
};
