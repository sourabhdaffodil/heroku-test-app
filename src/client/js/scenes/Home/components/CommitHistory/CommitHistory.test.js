import React from 'react';
import { shallow } from 'enzyme';
import { CommitHistory, mapStateToProps } from './CommitHistory';

describe('Commit history component', () => {
  const commits = {
    2016: [
      {
        _id: '59d678d9f51fea02ea1209b7',
        date: '2016-06-04T23:00:00.000Z',
        title: 'A title',
        description: 'Description...',
        image: 'https://avatars0.githubusercontent.com/u/4216535?v=4&s=72',
        hash: 'd9094385acf7d47c7c5c6f7ceb6866b54f9b602d',
        hashAlias: 'd909438',
      },
    ],
    2015: [
      {
        _id: '59d678d9f51fea02ea1209b7',
        date: '2015-06-04T23:00:00.000Z',
        title: 'A title',
        description: 'Description...',
        image: 'https://avatars0.githubusercontent.com/u/4216535?v=4&s=72',
        hash: 'd9094385acf7d47c7c5c6f7ceb6866b54f9b602d',
        hashAlias: 'd909438',
      },
    ],
  };

  it('Must render an empty list properly', () => {
    const container = shallow(
      <CommitHistory commits={{}} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('Must render a list of commits', () => {
    const wrapper = shallow(
      <CommitHistory commits={commits} />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  describe('Commit history mapStateToProps function', () => {
    it('Must return an object with an empty "commits" property', () => {
      const defaultState = { history: { commits: {} } };
      const result = mapStateToProps(defaultState);
      const expected = { commits: {} };
      expect(result).toEqual(expected);
    });

    it('Must map the state "commits" to the component\'s properties', () => {
      const result = mapStateToProps({ history: { commits } });
      const expected = {
        commits,
      };
      expect(result).toEqual(expected);
    });
  });
});
