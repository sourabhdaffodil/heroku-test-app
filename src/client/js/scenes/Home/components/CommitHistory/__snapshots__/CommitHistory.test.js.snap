// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commit history component Must render a list of commits 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommitHistory
    commits={
      Object {
        "2015": Array [
          Object {
            "_id": "59d678d9f51fea02ea1209b7",
            "date": "2015-06-04T23:00:00.000Z",
            "description": "Description...",
            "hash": "d9094385acf7d47c7c5c6f7ceb6866b54f9b602d",
            "hashAlias": "d909438",
            "image": "https://avatars0.githubusercontent.com/u/4216535?v=4&s=72",
            "title": "A title",
          },
        ],
        "2016": Array [
          Object {
            "_id": "59d678d9f51fea02ea1209b7",
            "date": "2016-06-04T23:00:00.000Z",
            "description": "Description...",
            "hash": "d9094385acf7d47c7c5c6f7ceb6866b54f9b602d",
            "hashAlias": "d909438",
            "image": "https://avatars0.githubusercontent.com/u/4216535?v=4&s=72",
            "title": "A title",
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <styled.div>
          <styled.div>
            <Icon
              icon="circle-o"
            />
            <styled.div>
              <span>
                Commits on 
              </span>
              2016
            </styled.div>
          </styled.div>
          <CommitGroup
            commits={
              Array [
                Object {
                  "_id": "59d678d9f51fea02ea1209b7",
                  "date": "2016-06-04T23:00:00.000Z",
                  "description": "Description...",
                  "hash": "d9094385acf7d47c7c5c6f7ceb6866b54f9b602d",
                  "hashAlias": "d909438",
                  "image": "https://avatars0.githubusercontent.com/u/4216535?v=4&s=72",
                  "title": "A title",
                },
              ]
            }
          />
        </styled.div>,
        <styled.div>
          <styled.div>
            <Icon
              icon="circle-o"
            />
            <styled.div>
              <span>
                Commits on 
              </span>
              2015
            </styled.div>
          </styled.div>
          <CommitGroup
            commits={
              Array [
                Object {
                  "_id": "59d678d9f51fea02ea1209b7",
                  "date": "2015-06-04T23:00:00.000Z",
                  "description": "Description...",
                  "hash": "d9094385acf7d47c7c5c6f7ceb6866b54f9b602d",
                  "hashAlias": "d909438",
                  "image": "https://avatars0.githubusercontent.com/u/4216535?v=4&s=72",
                  "title": "A title",
                },
              ]
            }
          />
        </styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "2016",
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div>
              <Icon
                icon="circle-o"
              />
              <styled.div>
                <span>
                  Commits on 
                </span>
                2016
              </styled.div>
            </styled.div>,
            <CommitGroup
              commits={
                Array [
                  Object {
                    "_id": "59d678d9f51fea02ea1209b7",
                    "date": "2016-06-04T23:00:00.000Z",
                    "description": "Description...",
                    "hash": "d9094385acf7d47c7c5c6f7ceb6866b54f9b602d",
                    "hashAlias": "d909438",
                    "image": "https://avatars0.githubusercontent.com/u/4216535?v=4&s=72",
                    "title": "A title",
                  },
                ]
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Icon
                  icon="circle-o"
                />,
                <styled.div>
                  <span>
                    Commits on 
                  </span>
                  2016
                </styled.div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "icon": "circle-o",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <span>
                      Commits on 
                    </span>,
                    "2016",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Commits on ",
                    },
                    "ref": null,
                    "rendered": "Commits on ",
                    "type": "span",
                  },
                  "2016",
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "commits": Array [
                Object {
                  "_id": "59d678d9f51fea02ea1209b7",
                  "date": "2016-06-04T23:00:00.000Z",
                  "description": "Description...",
                  "hash": "d9094385acf7d47c7c5c6f7ceb6866b54f9b602d",
                  "hashAlias": "d909438",
                  "image": "https://avatars0.githubusercontent.com/u/4216535?v=4&s=72",
                  "title": "A title",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2015",
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div>
              <Icon
                icon="circle-o"
              />
              <styled.div>
                <span>
                  Commits on 
                </span>
                2015
              </styled.div>
            </styled.div>,
            <CommitGroup
              commits={
                Array [
                  Object {
                    "_id": "59d678d9f51fea02ea1209b7",
                    "date": "2015-06-04T23:00:00.000Z",
                    "description": "Description...",
                    "hash": "d9094385acf7d47c7c5c6f7ceb6866b54f9b602d",
                    "hashAlias": "d909438",
                    "image": "https://avatars0.githubusercontent.com/u/4216535?v=4&s=72",
                    "title": "A title",
                  },
                ]
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Icon
                  icon="circle-o"
                />,
                <styled.div>
                  <span>
                    Commits on 
                  </span>
                  2015
                </styled.div>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "icon": "circle-o",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <span>
                      Commits on 
                    </span>,
                    "2015",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Commits on ",
                    },
                    "ref": null,
                    "rendered": "Commits on ",
                    "type": "span",
                  },
                  "2015",
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "function",
            "props": Object {
              "commits": Array [
                Object {
                  "_id": "59d678d9f51fea02ea1209b7",
                  "date": "2015-06-04T23:00:00.000Z",
                  "description": "Description...",
                  "hash": "d9094385acf7d47c7c5c6f7ceb6866b54f9b602d",
                  "hashAlias": "d909438",
                  "image": "https://avatars0.githubusercontent.com/u/4216535?v=4&s=72",
                  "title": "A title",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <styled.div>
            <styled.div>
              <Icon
                icon="circle-o"
              />
              <styled.div>
                <span>
                  Commits on 
                </span>
                2016
              </styled.div>
            </styled.div>
            <CommitGroup
              commits={
                Array [
                  Object {
                    "_id": "59d678d9f51fea02ea1209b7",
                    "date": "2016-06-04T23:00:00.000Z",
                    "description": "Description...",
                    "hash": "d9094385acf7d47c7c5c6f7ceb6866b54f9b602d",
                    "hashAlias": "d909438",
                    "image": "https://avatars0.githubusercontent.com/u/4216535?v=4&s=72",
                    "title": "A title",
                  },
                ]
              }
            />
          </styled.div>,
          <styled.div>
            <styled.div>
              <Icon
                icon="circle-o"
              />
              <styled.div>
                <span>
                  Commits on 
                </span>
                2015
              </styled.div>
            </styled.div>
            <CommitGroup
              commits={
                Array [
                  Object {
                    "_id": "59d678d9f51fea02ea1209b7",
                    "date": "2015-06-04T23:00:00.000Z",
                    "description": "Description...",
                    "hash": "d9094385acf7d47c7c5c6f7ceb6866b54f9b602d",
                    "hashAlias": "d909438",
                    "image": "https://avatars0.githubusercontent.com/u/4216535?v=4&s=72",
                    "title": "A title",
                  },
                ]
              }
            />
          </styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "2016",
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div>
                <Icon
                  icon="circle-o"
                />
                <styled.div>
                  <span>
                    Commits on 
                  </span>
                  2016
                </styled.div>
              </styled.div>,
              <CommitGroup
                commits={
                  Array [
                    Object {
                      "_id": "59d678d9f51fea02ea1209b7",
                      "date": "2016-06-04T23:00:00.000Z",
                      "description": "Description...",
                      "hash": "d9094385acf7d47c7c5c6f7ceb6866b54f9b602d",
                      "hashAlias": "d909438",
                      "image": "https://avatars0.githubusercontent.com/u/4216535?v=4&s=72",
                      "title": "A title",
                    },
                  ]
                }
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Icon
                    icon="circle-o"
                  />,
                  <styled.div>
                    <span>
                      Commits on 
                    </span>
                    2016
                  </styled.div>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "icon": "circle-o",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <span>
                        Commits on 
                      </span>,
                      "2016",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Commits on ",
                      },
                      "ref": null,
                      "rendered": "Commits on ",
                      "type": "span",
                    },
                    "2016",
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "commits": Array [
                  Object {
                    "_id": "59d678d9f51fea02ea1209b7",
                    "date": "2016-06-04T23:00:00.000Z",
                    "description": "Description...",
                    "hash": "d9094385acf7d47c7c5c6f7ceb6866b54f9b602d",
                    "hashAlias": "d909438",
                    "image": "https://avatars0.githubusercontent.com/u/4216535?v=4&s=72",
                    "title": "A title",
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2015",
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div>
                <Icon
                  icon="circle-o"
                />
                <styled.div>
                  <span>
                    Commits on 
                  </span>
                  2015
                </styled.div>
              </styled.div>,
              <CommitGroup
                commits={
                  Array [
                    Object {
                      "_id": "59d678d9f51fea02ea1209b7",
                      "date": "2015-06-04T23:00:00.000Z",
                      "description": "Description...",
                      "hash": "d9094385acf7d47c7c5c6f7ceb6866b54f9b602d",
                      "hashAlias": "d909438",
                      "image": "https://avatars0.githubusercontent.com/u/4216535?v=4&s=72",
                      "title": "A title",
                    },
                  ]
                }
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Icon
                    icon="circle-o"
                  />,
                  <styled.div>
                    <span>
                      Commits on 
                    </span>
                    2015
                  </styled.div>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "icon": "circle-o",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <span>
                        Commits on 
                      </span>,
                      "2015",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Commits on ",
                      },
                      "ref": null,
                      "rendered": "Commits on ",
                      "type": "span",
                    },
                    "2015",
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "function",
              "props": Object {
                "commits": Array [
                  Object {
                    "_id": "59d678d9f51fea02ea1209b7",
                    "date": "2015-06-04T23:00:00.000Z",
                    "description": "Description...",
                    "hash": "d9094385acf7d47c7c5c6f7ceb6866b54f9b602d",
                    "hashAlias": "d909438",
                    "image": "https://avatars0.githubusercontent.com/u/4216535?v=4&s=72",
                    "title": "A title",
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Commit history component Must render an empty list properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommitHistory
    commits={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
